### PR TITLE
Build tool: Generate the release notes programatically by taking all the information from the pull request titles

### PR DESCRIPTION
**Pull request by hmaarrfk**
Takes all the info from Github PRs

releates to:
https://github.com/scikit-image/scikit-image/issues/3404

Generate yourself a public repo github token to test this.

Install
PyGithub and tqdm (only one of the lines below).
```
conda install pygithub tqdm
pip install pygithub tqdm
```

```
python generate_release_notes.py
```

and 
```
python generate_release_notes.py > my_release.rst
```

~~That said, it seems a little buggy for now.~~

~~Known issues:~~
- ~~This searches for PRs by date. This kinda worked when the process was linear, but doesn't work so well now that we have 2 branches. Instead, we should search if a commit intersects the current (or desired) branch.~~
- ~~The committers finds the commiters on the current branch assuming that the currnet branch interects the tag?~~

## Description
[Tell us about your new feature, improvement, or fix! If relevant, please supplement the PR with images.]

## Checklist
[It's fine to submit PRs which are a work in progress! But before they are merged, all PRs should provide:]
- [ ] Clean style in [the spirit of PEP8](https://www.python.org/dev/peps/pep-0008/)
- [ ] [Docstrings for all functions](https://github.com/numpy/numpy/blob/master/doc/example.py)
- [ ] Gallery example in `./doc/examples` (new features only)
- [ ] Benchmark in `./benchmarks`, if your changes aren't covered by an
  existing benchmark
- [ ] Unit tests

[For detailed information on these and other aspects see [scikit-image contribution guidelines](http://scikit-image.org/docs/dev/contribute.html)]

## References
[If this is a bug-fix or enhancement, it closes issue # ]
[If this is a new feature, it implements the following paper: ]

## For reviewers

(Don't remove the checklist below.)

- [ ] Check that the PR title is short, concise, and will make sense 1 year
  later.
- [ ] Check that new functions are imported in corresponding `__init__.py`.
- [ ] Check that new features, API changes, and deprecations are mentioned in
      `doc/release/release_dev.rst`.
- [ ] Consider backporting the PR with `@meeseeksdev backport to v0.14.x`

### CODE REVIEW

**Comment by jni** [comment]
@hmaarrfk this is very cool, and it's great that it goes by branch rather than date, but it seems to miss a lot of contributions. Maybe you're only walking one branch when there's a merge? Contrast the contributors list in my [manually curated list](https://github.com/scikit-image/scikit-image/blob/ab4c289654b9f22c696a642c7c929fe046bd305b/doc/release/release_0.14.rst) to the one [output](https://gist.github.com/jni/adede2a910cd60974a2568c5ef6f99d4) by your tool.

I also think, long term, we want the release notes to point to the "canonical" PR, rather than the backported one. That might be harder or even impossible, though.

**Comment by hmaarrfk** [comment]
The only PRs that don't appear as numbers on the list are:
- #3313 which is listed as #3314
-  #3391 which was folded into #3052

Both of those are actually the relevant backports.

I think we can make them point to the canonical PR. I was just too lazy to implement this for a "proof of concept" For example, for `Backport PR #3374 on branch v0.14.x ` we could find the 3374 and look up that PR and get the title. I mostly just hate regex. These scripts require some buy in, and "automagic" stuff isn't always easy to sell.

What I really don't know how to do, is how to find the PRs that were already in 0.14.1 when 0.15 rolls around.

**Comment by jni** [comment]
Sorry: I meant contribuTORS.

**Comment by hmaarrfk** [comment]
Oh right. Not too sure how to handle that one. WIP

**Comment by jni** [comment]
> What I really don't know how to do, is how to find the PRs that were already in 0.14.1 when 0.15 rolls around.

We could tag branch points: whatever the base commit for the v0.14.x branch is, tag as, e.g. v0.14-branch. Then you search PRs in between v0.15-branch and v0.14-branch.

> Not too sure how to handle that one. [contributors]

I presume the error is because you are only looking at one of the two parents of merge commits, and you end up only examining merge commits and none of the actual contributed commits? I've never played with the git graph but we need to make sure that we look at *all* paths from v0.14.0 and v0.14.1, for example. If I have to cast the git DAG into networkx to do this easily, I will. =P

> I was just too lazy to implement this for a "proof of concept"

Understood, but I'm pretty impressed with your proof. =) I definitely want to use it (or a refined version of it) in the future! =)

**Comment by hmaarrfk** [comment]
> We could tag branch points: whatever the base commit for the v0.14.x branch is, tag as, e.g. v0.14-branch. Then you search PRs in between v0.15-branch and v0.14-branch.

It handles this nicely. but then the announcement would be: "these are the changes between 0.15.0 and 0.14.0" when people will be more interested in "changes between 0.15.0 and 0.14.1"

**Comment by hmaarrfk** [comment]
Most of this builds off the previous `contrib.py` work. None of this was my invention per say.

**Comment by jni** [comment]
> people will be more interested in "changes between 0.15.0 and 0.14.1"

AAAAAHHHH, this is a very good point! This LTS stuff is going to be tricky. Later on, will people be interested in the changes between 0.15 and 0.16, or 0.14.7 and 0.16? :face_with_head_bandage:

**Comment by hmaarrfk** [comment]
in my earlier comments, this is what I meant by just reverting our stance on 2.7. The challenges involved in maintaining two code paths is quite large :/ and I think we are only starting to see them.

**Comment by jni** [comment]
In which direction do you want to revert it? Both are horrible (for me).

Anyway, as I've thought more about it, I think it's not so bad. 0.14.(n+1) releases will want to note the changes since 0.14.n. 0.15(+n+1) releases will want to know the changes since 0.15+n. It's only this one time that people will have updated to 0.14.x who haven't updated to the next "major" version because there isn't one. Once people are on 0.15, they won't care what's going on with 0.14.x.

**Comment by jni** [comment]
Yes I've thought about it, but I'm gonna take a week's rest, then I'll consider it again. =)

**Comment by hmaarrfk** [comment]
@jni this script is up and working again ;)

**Comment by hmaarrfk** [comment]
<details> <summary> Sample output </summary>

```
Announcement: scikit-image 0.15.0
=================================

We're happy to announce the release of scikit-image v0.15.0!

scikit-image is an image processing toolbox for SciPy that includes algorithms
for segmentation, geometric transformations, color space manipulation,
analysis, filtering, morphology, feature detection, and moreself.


For more information, examples, and documentation, please visit our website:

http://scikit-image.org



Improvements
------------

- MNT: handle a deprecation warning for np.linspace and floats for the num parameter (#3453)
- TST: numpy empty arrays are not inherently Falsy (#3455)
-  handle warning in scipy cdist for unused parameters (#3456)
- MNT: don't use filter_warnings in test suite. (#3459)
- Add doc notes on setting up the build environment (#3472)
- Release the GIL in numerous cython functions (#3490)
- Cython touchups to use float32 and float64 (#3493)
- rank_filters: Change how the bitdepth and max_bin are computed to ensure exact warnings. (#3501)
- Early stop rank kernel noise filter as intended (#3503)
- RANK: Optimize OTSU filter (#3504)
- Rank - Fix rank entropy and OTSU tests (#3506)
- delay importing pyplot in manual segmentation (#3533)
- Get rid of the requirements-parser dependency (#3534)
- filter warning from `correct_mesh_orientation` in tests (#3549)
- cloudpickle is really a doc dependency, not a core one (#3634)
- optional dependencies on pip (#3645)
- Fewer test warnings in 3.7 (#3687)
- collections.abc nit (#3692)
- Streamlined issue template (#3697)
- Tighten the PR Template (#3701)
- Use language level to 3 in cython for future compatibility (#3707)
- Update ISSUE_TEMPLATE.md with info about numpy and skimage versions (#3730)
- Use relative imports for many cython modules (#3759)

Bugfixs
-------

- an index needs to be an integer not an float in CollectionViewer (#3288)
- erroneous use of cython wrap around (#3481)
- handle correctly the maximum number of lines in hough transform (#3514)
- Speed up block reduce by providing the appropriate parameters to numpy (#3522)
- Add urllib.request again (#3766)

Build Tools
-----------

- Pass tests that don't raise floating point exceptions on arm with soft-fp (#3337)

Other Pull Requests
-------------------

- ENH: add range option for histogram. (#2479)
- BUG: Fix greycoprops correlation always returning 1 (#2532)
- Add section on API discovery via `skimage.lookfor` (#2539)
- Attribute operators (#2680)
- Better document remove_small_objects behaviour: int vs bool (#2830)
- Extension of register_translation to enable subpixel precision in 3D and optionally disable error calculation (#2880)
- Speedup 2D warping for affine transformations (#2902)
- Credit Reviewers in Release Notes (#2927)
- Added small galleries in the API (#2940)
- Use skimage gaussian filter to avoid integer rounding artifacts (#2983)
- Remove Python 2 compatibility (#3000)
- Replace morphology.local_maxima with faster flood-fill based Cython version (#3022)
- Allow float->float conversion of any range (#3052)
- Add `rectangle_perimeter` feature to `skimage.draw` (#3069)
- Use proper axis indexing in `adapt_rgb` `each_channel` (#3097)
- Linking preserve_range parameter calls to docs (#3109)
- Let lower precision float arrays pass through `img_as_float` (#3110)
- Update installation instructions to reference existing requirements specification (#3113)
- Updated release notes with pre 0.13.1 phase (#3114)
- Release guidelines update (#3115)
- Ensure we are installing with / running on Python 3 (#3119)
- Lazy apply_parallel (#3121)
- Prefer imageio over PIL for file manipulations. (#3126)
- Update the documentation regarding datalocality (#3127)
- Hide warnings in test_unsharp_mask (#3130)
- Process 0.15 deprecations (#3132)
- Documentation: always use dev branch javascript (#3136)
- Add initial airspeed velocity (asv) framework (#3137)
- Supress warnings for flatten during io testing (#3143)
- Bugfix separate stains log10 (#3146)
- Recover from exceptions in filters.try_all_threshold() (#3149)
- Fix  skimage.test() to run the unittests (#3152)
- skivi: Use qtpy to handle different Qt versions (#3157)
- Refractor python version checking. (#3160)
- moved data_dir to within `data/__init__.py` (#3161)
- Moved the definition of lookfor out of __init__.py (#3162)
- Normalized the package number to PEP440 (#3163)
- Remove skimage.test as it was never used. (#3164)
- Added a message about qtpy to the INSTALL.rst (#3168)
- Regression fix: Travis should fail if tests fail (#3170)
- Set minimum cython version to `0.23.4` (#3171)
- Repeat pixels in reflect mode when image has dimension 1 (#3174)
- Add rgba2rgb to API docs (#3175)
- Minor docs formatting fixes in video.rst (#3176)
- Decrease the verbosity of the testing (#3182)
- Speedup rgb2gray using matrix multiply (#3187)
- Update docs: specify conda-forge channel for scikit-image conda install (#3189)
- Added instructions for meeseeksdev to PR template (#3194)
- Remove installation instructions for video packages (#3197)
- Big image labeling fix (#3202)
- Handle dask deprecation in cycle_spin (#3205)
- Fix Qt viewer painttool indexing (#3210)
- build_versions.py is no longer hard coded. (#3211)
- Removed dtype constructor call in exposure.rescale_intensity (#3213)
- Various updates to the ASV benchmarks (#3215)
- Add a link to stack overflow on github README (#3217)
- MAINT: remove encoding information in file headers (python 3) (#3219)
- Build tools: Dedicate a --pre build in appveyor and ensure other builds don't download --pre (#3222)
- Fix the human readable error message on a bad build. (#3223)
- Respect input array type in apply_parallel by default (#3225)
- Travis cleanup pip commands (#3227)
- Add benchmarks for morphology.watershed (#3234)
- Corrected docstring formatting so that code block is displayed as code (#3236)
- numpy: Handle multidimentional indexing in 1.15 (#3238)
- pytest: Numpy 1 15 warnings (#3242)
- FEAT: Flood fill in n-dimensions (#3245)
- Defer skimage.io import of matplotlib.pyplot until needed (#3243)
- Add benchmark for Sobel filters (#3249)
- Remove cython md5 hashing since it breaks the build process (#3254)
- Fix typo in documentation. (#3262)
- Issue 3156: skimage/__init__.py Update docstring and fix import *  (#3265)
- Object detector module (#3267)
- Do not import submodules while building (#3270)
- Add benchmark suite for canny (#3271)
- improve segmentation.felzenszwalb document #3264 (#3272)
- Update _canny.py (#3276)
- Fix background indexing in _label2rgb_avg (#3280)
- Fix skimage.measure.regionprops 1x1 image bug  (#3284)
- Add benchmark suite for histogram equalization (#3285)
- fix link to equalist_hist blog reference (#3287)
- .gitignore: novice: Ignore save-demo.jpg (#3289)
- Guide the user of denoise_wavelet to choose an orthogonal wavelet. (#3290)
- Remove unused lib in skimage/__init__.py (#3291)
- Replace scipy.sparse.*.todense() with toarray() (#3292)
- BUILD: Add pyproject.toml to ensure cython is present (#3295)
- Handle intersphinx and mpl deprecation warnings in docs (#3300)
- Fix incorrect inertia tensor (#3303)
- Minor PEP8 fixes (#3305)
- cython: check for presence of cpp files during install from sdist (#3311)
- Remove deprecated `dynamic_range` in `measure.compare_psnr` (#3313)
- appveyor: don't upload any artifacts (#3315)
- Add benchmark suite for hough_line() (#3319)
- Novice skip url test (#3320)
- Remove benchmarks from wheel (#3321)
- Add license file to the wheel (binary) distribution (#3322)
- codecov: ignore build scripts in coverage and don't comment on PRs (#3326)
- Implementation of masked image translation registration (#3334)
- Correct rotate method's center parameter doc (#3341)
- Matplotlib 2.2.3 +  PyQt5.11 (#3345)
- Allow @hmaarrfk to mention MeeseeksDev to backport. (#3357)
- Add Python 3.7 to the test matrix (#3359)
- Fix deprecated keyword from dask (#3366)
- Turn DOIs into web links in docstrings (#3367)
- Incompatible modes with anti-aliasing in skimage.transform.resize (#3368)
-  Missing cval parameter in threshold_local (#3370)
- add comma to measure init (#3374)
- Speed up inner loop of GLCM (#3378)
- Avoid Sphinx 1.7.8 (#3381)
- Show our data in the gallery (#3388)
- Minor updates to grammar in numpy images page (#3389)
- assert_all_close doesn't exist, make it `assert_array_equal` (#3391)
- Better behavior of Gaussian filter for arrays with a large number of dimensions (#3394)
- Allow import/execution with -OO (#3398)
- Mark tests known to fail on 32bit architectures with xfail (#3399)
- Improve Li thresholding (#3402)
- Hardcode the inputs to test_ssim_grad (#3403)
- TST: make test_wavelet_denoising_levels compatible with PyWavelets 1.0 (#3406)
- Allow tifffile.py to handle I/O. (#3409)
- Add explicit Trove classifier for Python 3 (#3415)
- Fix error in contribs.py (#3418)
- MAINT: remove pyside restriction since we don't support Python 3.4 anymore (#3421)
- Build tools: simplify how MPL_DIR is obtained. (#3422)
- Build tools: Don't run tests twice in travis. (#3423)
- Build tools: Add an OSX build with optional dependencies. (#3424)
- MAINT: Reverted the changes in #3300 that broke the MINIMIUM_REQUIREMENTS tests (#3427)
- MNT: Convert links using http to https (#3428)
- MAINT: Use upstream colormaps now that matplotlib has been upgraded (#3429)
- Build tools: Make pyamg an optional dependency and remove custom logic (#3431)
- Build tools: Fix PyQt installed in minimum requirements build (#3432)
- MNT: multiprocessing should always be avaible since we depend on python >=2.7 (#3434)
- MAINT Use np.full instead of cst*np.ones (#3440)
-  DOC: Fix LaTeX build via `make latexpdf`  (#3441)
- Update instructions et al for releases after 0.14.1 (#3442)
- Remove code specific to python 2 (#3443)
- Fix default value of `methods` in `_try_all` to avoid exception (#3444)
-  Fix morphology.local_maxima for input with any dimension < 3 (#3447)
- Use raw strings to avoid unknown escape symbol warnings (#3450)
- Speed up xyz2rgb by clipping output in place (#3451)
- MNT; handle deprecation warnings in tifffile (#3452)
- Build tools: TST: filter away novice deprecation warnings during testing (#3454)
- Build tools: don't use the pytest.fixtures decorator anymore in class fixtures  (#3458)
- Preserving the fill_value of a masked array (#3461)
- Fix VisibleDeprecationWarning from np.histogram, normed=True (#3463)
- Build Tools: DOC: Document that now PYTHONOPTMIZE build is blocked by SciPy (#3470)
- DOC: Replace broken links by webarchive equivalent links (#3471)
- FIX: making the plot_marching_cubes example visible. (#3474)
- Avoid Travis failure regarding ``skimage.lookfor`` (#3477)
- Fix Python executable for sphinx-build in docs Makefile (#3478)
- Build Tools: Blacklist specific Cython versions (#3479)
- Fix typos (#3480)
- Add "optional" indications to docstrings (#3495)
- Rename 'mnxc' (masked normalize cross-correlation) to something more descriptive (#3497)
- Random walker bug fix: no error should be raised when there is nothing to do (#3500)
- Various minor edits for active contour (#3508)
- Fix range for uint32 dtype in user guide (#3512)
- Raise meaningful exception in warping when image is empty (#3518)
- DOC: Development installation instructions for Ubuntu are missing tkinter (#3520)
- Better gallery examples and tests for masked translation registration (#3528)
- DOC: make more docstrings compliant with our standards (#3529)
- Add copybutton (#3530)
- Build tools: Remove restriction on simpleitk for python 3.7 (#3535)
- Speedup and add benchmark for `skeletonize_3d` (#3536)
- Update requirements/README.md on justification of matplotlib 3.0.0 in favor of #3476 (#3542)
- Doc enhancements around denoising features. (#3553)
- DOC: Improve the RANSAC gallery example (#3554)
- Handle deprecation of numpy `_validate_lengths` (#3556)
- Use 'getconf _NPROCESSORS_ONLN' as fallback for nproc in Makefile of docs (#3563)
- Fix matplotlib set_*lim API deprecations (#3564)
- Add histogram matching (#3568)
- Switched from np.power to np.cbrt (#3570)
- Filtered out DeprecationPendingWarning for matrix subclass (#3572)
- Add RGB to grayscale example to gallery (#3574)
- Build tools: Refactor check_sdist so that it takes a filename as a parameter (#3579)
- Turn dask to an optional requirement (#3582)
- _marching_cubes_lewiner_cy: mark char as signed (#3587)
- Hyperlink DOIs to preferred resolver (#3589)
- Missing parameter description in `morphology.reconstruction() docstring #3581 (#3591)
- EHN: allow tuple to define kernel in threshold_niblack and threshold_sauvola (#3596)
- Update chat location (#3598)
- DOC: "feature.peak_local_max" : explanation of multiple same-intensity peaks returned by the function; added details on `exclude_border` parameter  (#3600)
- Removed orphan code (skimage/filters/_ctmf.pyx). (#3601)
- [MRG] update documentation for regionprops (#3602)
- More explicit example title, better list rendering in plot_cycle_spinning.py (#3606)
- Add rgb to hsv example in the gallery (#3607)
- [MRG] Update documentation of `perimeter` and add input validation (#3608)
- Additionnal mask option to clear_border (#3610)
- Set up CI with Azure Pipelines (#3612)
- [MRG] EHN: median filters will accept floating image (#3616)
- Update Travis-CI to xcode 10.1 (#3617)
- Minor tweaks to _mean_std code (#3619)
- Added glossary to the doc (#3626)
- Add explicit ordering of gallery sections (#3627)
- Delete broken links (#3628)
- Build tools: Fix test_mpl_imshow for matplotlib 2.2.3 and numpy 1.16 (#3635)
- First draft of core dev guide (#3636)
- Add more details about the home page build process (#3639)
- Ensure images resources with long querystrings can be read (#3642)
- Delay matplotlib import in skimage/future/manual_segmentation.py (#3648)
- make the low contrast check optional when saving images (#3653)
- Correctly ignore release notes auto-generated for docs (#3656)
- Remove MANIFEST file when making the 'clean' target (#3657)
- Clarify return values in _overlap docstrings in feature/blob.py (#3660)
- Contribution script: allow specification of GitHub development branch (#3661)
- Update core dev guide: deprecation, contributor guide, required experience (#3662)
- Add release notes for 0.14.2 (#3664)
- FIX gallery: Add multichannel=True to match_histogram (#3672)
- MAINT Minor code style improvements (#3673)
- Pass parameters through tifffile plugin (#3675)
- DOC unusused im3d_t in example (#3677)
- Remove wrong cast of Py_ssize_t to int (#3682)
- Build tools: allow python 3.7 to fail, but travis to continue (#3683)
- Build tools: remove pyproject.toml (#3688)
- Add support for anisotropic blob detection in blob_log and blob_dog (#3690)
- Fix ValueError: not enough values to unpack (#3703)
- ENH: several fixes for heap.pyx (#3704)
- Enable the faulthandler module during testing (#3708)
- Build tools: Fix Python 3.7 builds on travis (#3709)
- Replace np.einsum with np.tensordot in _upsampled_dft (#3710)
- Fix potential use of NULL pointers (#3717)
- Fix potential memory leak (#3718)
- Fix potential use of NULL pointers (#3719)
- Fix and improve core_cy.pyx (#3720)
- Build tools: Downgrade Xcode to 9.4 on master (#3723)
- Improve visual_test.py (#3732)
- Updated painttool to work with color images and properly scale labels. (#3733)
- Add image.sc forum badge to README (#3738)
- Blacklist PyQt 5.12.0 on Travis (#3743)
- Build tools: Fix matplotlib + qt 5.12 the same way upstream does it (#3744)
- Add image of retina to our data (#3748)
- Make local_maxima return a boolean array (#3749)
- Remove seam carving (#3751)
- gallery: remove xx or yy  sorted directory names (#3761)
- Allow for f-contiguous 2D arrays in convex_hull_image (#3762)
- Add microaneurysms() to gallery (#3765)
- Build tools: Set astropy minimum requirement to 1.2 to help the CIs. (#3767)
- Avoid NumPy warning while stacking arrays. (#3768)

44 authors added to this release [alphabetical by last name]
------------------------------------------------------------
- Abhishek Arya
- Tom Augspurger
- Marcel Beining
- François Boulogne
- Genevieve Buckley
- Matthias Bussonnier
- François COKELAER
- Ambrose J Carr
- Zhanwen "Phil" Chen
- Casper da Costa-Luis
- Laurent P. René de Cotret
- Sarkis Dallakian
- Christoph Deil
- Jan Eglinger
- Kira Evans
- Christoph Gohlke
- Emmanuelle Gouillart
- Lars Grueter
- Mark Harfouche
- Jonathan J. Helmus
- Oscar Javier Hernandez
- Hugo
- Arthur Imbert
- Jan
- Jathrone
- Justin
- KimNewell
- Kirill Klimov
- Gregory R. Lee
- Legodev
- Katrin Leinweber
- Guillaume Lemaitre
- Tommy Löfstedt
- Matt McCormick
- Jeremy Metz
- Juan Nunez-Iglesias
- Onomatopeia
- Thein Oo
- Page-David
- Egor Panfilov
- Jesse Pangburn
- Constantin Pape
- PeterJackNaylor
- PinkFloyded
- François-Michel De Rainville
- Christian Rauch
- Rob
- Adrian Roth
- Johannes Schönberger
- Taylor D. Scott
- Brian Smith
- Scott Staniewicz
- Tony Tung
- Nelle Varoquaux
- Vilim Štih
- Stefan van der Walt
- Thomas Walter
- Nehal J Wani
- Josh Warner
- Roman Yurchak
- alexis-cvetkov
- azure-pipelines[bot]
- blochl
- fivemok
- jakirkham
- mamrehn
- mrastgoo
- ratijas
- robroooh
- seanbudd
- yangfl


44 committers added to this release [alphabetical by last name]
---------------------------------------------------------------
- Abhishek Arya
- Tom Augspurger
- François Boulogne
- Genevieve Buckley
- Matthias Bussonnier
- Casper da Costa-Luis
- Laurent P. René de Cotret
- Christoph Deil
- Kira Evans
- Emmanuelle Gouillart
- Lars Grueter
- Mark Harfouche
- Jonathan J. Helmus
- Oscar Javier Hernandez
- Arthur Imbert
- Kirill Klimov
- Gregory R. Lee
- Katrin Leinweber
- Guillaume Lemaitre
- Tommy Löfstedt
- Juan Nunez-Iglesias
- Onomatopeia
- Thein Oo
- Egor Panfilov
- Constantin Pape
- PinkFloyded
- François-Michel De Rainville
- Christian Rauch
- Adrian Roth
- Johannes Schönberger
- Brian Smith
- Scott Staniewicz
- Tony Tung
- Vilim Štih
- Stefan van der Walt
- Thomas Walter
- Josh Warner
- Roman Yurchak
- alexis-cvetkov
- blochl
- jakirkham
- robroooh
- seanbudd
- yangfl


44 reviewers added to this release [alphabetical by last name]
--------------------------------------------------------------
- Jotham Apaloo
- Abhishek Arya
- Tom Augspurger
- Marcel Beining
- François Boulogne
- Matthew Brett
- François COKELAER
- Ambrose J Carr
- Thomas A Caswell
- Laurent P. René de Cotret
- Ilya Flyamer
- Christoph Gohlke
- Emmanuelle Gouillart
- Lars Grueter
- Mark Harfouche
- Arthur Imbert
- Justin
- Tomas Kazmar
- Gregory R. Lee
- Guillaume Lemaitre
- Evan Putra Limanto
- Tommy Löfstedt
- Jarrod Millman
- Juan Nunez-Iglesias
- Onomatopeia
- Thein Oo
- Egor Panfilov
- PeterJackNaylor
- Matti Picus
- Christian Rauch
- Rob
- Matthew Rocklin
- Adrian Roth
- Johannes Schönberger
- Alexandre de Siqueira
- Brian Smith
- Scott Staniewicz
- Grégory Starck
- Vilim Štih
- Stefan van der Walt
- Thomas Walter
- Josh Warner
- Roman Yurchak
- jakirkham
- mrastgoo

```

</details>

**Comment by hmaarrfk** [comment]
@scikit-image/core this script takes about 5 minutes to run, but I think if we work on things together, we can eventuallly sort through the 329 commits between 0.14.0 and 0.15.0 :D.

It simply requires renaming the PRs on github.

File: tools/generate_release_notes.py [hunk 1]
```diff
@@ -0,0 +1,215 @@
...
+announcement_title = "Announcement: scikit-image {}".format(version)
+print(announcement_title)
+print("="*len(announcement_title))
+
+print("""
+We're happy to announce the release of scikit-image v{version}!
+
+scikit-image is an image processing toolbox for SciPy that includes algorithms
+for segmentation, geometric transformations, color space manipulation,
+analysis, filtering, morphology, feature detection, and moreself.
```
Comment (stefanv) [inline tools/generate_release_notes.py]: moreself -> more

File: tools/generate_release_notes.py [hunk 2]
```diff
@@ -0,0 +1,215 @@
...
+""".format(version=version))
+
+if version.startswith('0.14'):
```
Comment (stefanv) [inline tools/generate_release_notes.py]: Remove 0.14 specific code

**Comment by jni** [comment]
@hmaarrfk just in case it wasn't clear, I'm working on this, please don't do any pushes in the next few hours. =)

**Comment by hmaarrfk** [comment]
In my mind, anything that gets 0.15 released is perfect ;)

File: tools/generate_release_notes.py (hunk 2 above)
Comment (hmaarrfk) [reply tools/generate_release_notes.py]: why? shouldn't this stay at least till the end of 2019?

**Comment by jni** [comment]
@stefanv this is ready for another review. Output here:

https://gist.github.com/jni/0d5b56b1d1a5a6baa2bfa03dad89bd08

There's still a few quirks, but I think this is a sufficiently good start for manual editing.

@hmaarrfk I removed a lot of the github user login checking because it was brittle (as you noted). So there are some duplicates now but they are readily removed by hand.

**Comment by stefanv** [comment]
Output looks good.  Once this is merged, we should open an issue to track the "Bugfixs" issue.

**Comment by stefanv** [comment]
Also let's list the other quirks in that issue, while they're fresh.

**Comment by stefanv** [comment]
LGTM; merging so that we can move ahead with 0.15.0.

**Pull request merged**